processor/src/main/java/org/mapstruct/ap/internal/conversion/AbstractStaticParseToStringConversion.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.internal.conversion;
+
+import org.mapstruct.ap.internal.model.common.ConversionContext;
+
+/**
+ * Base class for a conversion between a type <tt>T</tt> and {@link String},
+ * where {@code T#parse(String)} and {@code T#toString} are inverse operations.
+ */
+public abstract class AbstractStaticParseToStringConversion extends SimpleConversion {
+
+    @Override
+    protected String getToExpression(ConversionContext conversionContext) {
+        return "<SOURCE>.toString()";
+    }
+
+    @Override
+    protected String getFromExpression(ConversionContext conversionContext) {
+        return getParsingType().getCanonicalName() + ".parse( <SOURCE> )";
```

Comment:
We try to generate code that is human readable as much as possible. Instead of doing the FQN it would be better to use the short name if it is possible. Have a look at the `ConversionUtils`. `CurrencyToStringConversion` is also a nice example class for how this can look like.

Reply:
I did start with that, similar to `CurrencyToStringConversion` as you mention. Later, I added the base class as I noticed the common implementation between the three new conversions. The problem I ran into was [typeReferenceName](https://github.com/mapstruct/mapstruct/blob/cf668bea77d5d48f34d32c229823a04dbd01f21d/processor/src/main/java/org/mapstruct/ap/internal/conversion/ConversionUtils.java#L55) is `private` but I wanted to stay generic in the base class.
Solution for now was to fully qualify the class, though I did want to revisit.

Reply:
We can make the `typeReferenceName` `public` no problem. Thanks for looking into it